pkl-core/src/main/java/org/pkl/core/runtime/TestResults.java

```diff
@@ -85,147 +56,225 @@ public void setErr(String err) {
     this.err = err;
   }
 
-  public static class TestResult {
-
-    private final String name;
-    private final List<Failure> failures = new ArrayList<>();
-    private final List<Error> errors = new ArrayList<>();
-    private boolean isExampleWritten = false;
+  public static class TestSectionResults {
+    public final TestSection name;
+    private final List<TestResult> results = new ArrayList<>();
+    private Error error;
```

Comment:
There can be multiple errors per section. Right now our test reports don't surface this, but they will once this is merged: https://github.com/apple/pkl/pull/628

Reply:
Yeah, that's also what I thought would eventually happen :)  I did considered making it multi-error but I'm not that good at "Pkl internals" (yet XXD). If I have time, I'll see if I can test this PR with #628 as well!

Reply:
Pkl is a lazy language--even if an element throws, it doesn't prevent Pkl from executing other elements.

This works, for example:

```pkl
import "pkl:test"

hidden foo = new {
  throw("uh oh")
  throw("uh oh again!")
  "i am okay"
}

result1 = test.catchOrNull(() -> foo[0])
result2 = test.catchOrNull(() -> foo[1])
result3 = test.catchOrNull(() -> foo[2])
```

This produces:

```
result1 = "uh oh"
result2 = "uh oh again!"
result3 = null
```

So, a single test can possibly have multiple thrown errors, and it fits into Pkl's model perfectly fine.

Reply:
Ah trueI Then, I should change the commit message:

> Similarly, errors will also count as "one test with one assertion failed" because an error prevents any tests in a section from being evaluated (that is, "the whole section fails" until the error is corrected and we can then evaluate the assertions in the tests).

Here's a proposal:
>  Similarly errors will also count as "one test with one assertion failed" because, at the moment, individual errors are not being surfaced  individually. Instead, only the first error is being surfaced and thus it masks the rest of the tests and assertions in the test section where it happens.

Is this new version more accurate?

Reply:
I've addressed the commit message with the new wording.